PowerEditor/src/NppCommands.cpp

```diff
@@ -373,8 +373,8 @@ void Notepad_plus::command(int id)
 		case IDM_EDIT_COPY_BINARY:
 		case IDM_EDIT_CUT_BINARY:
 		{
-			size_t textLen = _pEditView->execute(SCI_GETSELTEXT, 0, 0) - 1;
-			if (!textLen)
+			size_t textLen = _pEditView->execute(SCI_GETSELTEXT, 0, 0);
+			if (textLen < 1)
 				return;

```

Comment:
> if (textLen < 1) is a more general check than just comparing with zero

How come it's a more general check?

Reply:
Not for this case where `textLen` is of type `size_t`.

But this comment applies in the next case, `textlen` is declared as `auto`. The compiler actually deduces it to be `int` type. To test this, I put the `- 1` back on line 3163 but changed line 3164 to be `if (textLen < 1)`. That alternative change was able to resolve the issue reported in #11639 as well. So, if line 3164 had been coded as `if (textLen < 1)`, it would have prevented the #11639 issue from manifesting because it would be checking for both 0 (pre-Scintilla5) and -1 (Scintilla5). That's why I called it a more general check for that case. Clearly, that claim does not apply for the `size_t textLen' in this case since size_t is unsigned and will wrap around. Pardon my hasty copy-paste.